AwesomeTSProject/src/containers/AddCrypto/Index.tsx

```diff
@@ -1,17 +1,67 @@
-import {Modal, View, Pressable, Text} from 'react-native';
+import {Modal, View, Pressable, Alert} from 'react-native';
 import React, {useState} from 'react';
 import {
   CryptoInput,
   AddText,
-  backtext,
-  modaltext,
-  modbutton,
-  backbutton,
+  BackText,
+  OpenModalText,
+  ModalText,
 } from './styles';
+import {useDispatch, useSelector} from 'react-redux';
+import {AppDispatch, RootState} from '../../../store';
+import {GET_CRYPTOS} from '../../../store/reducer/RootReducer';
+import LocalCryptos from '../../../store/localAPI/local';
+import {cryptoType} from '../../../store/types/CryptoTypes';
+import {theme} from '../../utils/theme';
 
 const AddCrypto = () => {
   const [text, onChangeText] = useState('');
   const [modalVisible, setModalVisible] = useState(false);
+  const [borderColor, setBorderColor] = useState(theme.colors.gray);
+
+  const Cryptos = useSelector(
+    (state: RootState) => state.selectedCrypto.cryptos,
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  const findDuplicate = () => {
+    const duplicate = Cryptos.find(
+      elem => elem.name === text || elem.symbol === text,
+    );
+    if (duplicate) {
+      return true;
+    }
+  };
+
+  const findStock = () => {
+    const newCrypto = LocalCryptos.find(
+      elem => elem.name === text || elem.symbol === text,
+    );
+    if (newCrypto) {
+      return newCrypto;
+    }
+    return false;
+  };
+
+  const handleCrypto = () => {
```

Comment:
In this block of code there is a lot of redundancy, the validations can also be improved.